distributions/standalone/src/main/resources/MIGRATION.txt

```diff
@@ -7,6 +7,7 @@ Please check https://developer.mulesoft.com/docs/display/current/Mule+ESB for mo
 Migration changes from 3.9.x to 3.10.x
 MULE-14381: By default, the File Connector processes empty files. For disabling this behavior use the mule.transport.file.ignoreEmptyFiles=true system property.
 MULE-14507: The HTTP Requester now sends a zero content length in empty payloads scenarios. For disabling this behavior use the mule.http.client.avoidZeroContentLength=true system property.
+MULE-14601: The JSON Schema Validator now fails if it receives a payload with duplicate keys. For allowing duplicate keys use the mule.json.validator.allowDuplicateKeys=true system property.
```

Comment:
Why did you choose to disallow duplicate keys as a default? Does the Json schema spec have the concept of allowing/disallowing duplicate keys? If the schema doesn't forbid it by default, then the validator shouldn't either.  It's weird to me that you had to add that on the parser level instead of the validator itself. I want to be sure this is the right move, instead of simply adding a new parameter to the validator object to enable this strict check.

Reply:
AWAIK, Json Schema Validator works with valid json objects (in our code it receives a JsonNode that was already parsed). I've investigated the json schema api that we use and it only provides the STRICT_DUPLICATE_DETECTION feature to its JsonParser, so no way to configure it for the validator. Regarding the default behavior, the Json specification doesn't prohibit the duplicate keys, though it states they SHOULD be avoided. This specification also mentions the possible duplicate-keys treatments: 1) to remain only the last value, 2) to fail, 3)  to remain all the keys. With this change, by default we take the second approach, and applying the system property, the third one. I'll discard the first approach because it entails data loss.

Reply:
if the spec doesn't' disallowed it, then you cannot change this. You would be breaking backwards compatibility, since some json documents currently considered valid will suddenly start to fail. The validator cannot fail on a recommendation. You should add an optional parameter which specifies if this should validated or not. By default, the current behavior must be kept